### PR TITLE
refactor(decimal): get locale from i18n provider if not passed in props - FE-4318

### PR DESCRIPTION
**Pull request by mmaciorowski-sage**
### Proposed behaviour
Get locale from i18n provider if not passed in props.

### Current behaviour
Locale is provided only in props.

### Checklist

- [x] Commits follow our style guide
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [x] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Carbon implementation and Design System documentation are congruent

### Testing instructions

https://codesandbox.io/s/carbon-quickstart-forked-lsrqo?file=/src/index.js

### CODE REVIEW

File: src/components/decimal/decimal.component.js [hunk 1]
```diff
@@ -323,7 +330,7 @@ Decimal.defaultProps = {
   align: "right",
   precision: 2,
   allowEmptyValue: false,
-  locale: "en",
+  locale: undefined,
```
Comment (jamime) [inline src/components/decimal/decimal.component.js:333]:
```suggestion
```

I don't think this is necessary.

File: src/components/configurable-items/configurable-items.component.js [hunk 2]
```diff
@@ -20,8 +20,8 @@ let deprecatedWarnTriggered = false;
 class ConfigurableItems extends React.Component {
   static contextType = LocaleContext;
 
-  constructor(props) {
-    super(props);
+  constructor(props, context) {
+    super(props, context);
```
Comment (jamime) [inline src/components/configurable-items/configurable-items.component.js:24]: This should be in separate commit. Does this add any functionality or is it just code style?

File: src/components/decimal/decimal.spec.js [hunk 3]
```diff
@@ -35,6 +36,14 @@ function render(props = {}, renderer = mount) {
   renderer(<Decimal {...defaultProps} />);
 }
 
+function renderI18n(locale) {
```
Comment (jamime) [inline src/components/decimal/decimal.spec.js:39]: This could use existing `mount` function, then tests could use existing `value()` method.

File: src/components/switch/switch.stories.mdx [hunk 4]
```diff
@@ -367,24 +367,24 @@ It is possible to use the `tooltipPosition` to override the default placement of
...
-      },
-      {
-        name: "switch.off",
-        description: "The text to display to indicate action would move component to off state",
-        type: "func",
-        returnType: "string"
-      },
-    ]
-  }
+  translationData={[
```
Comment (jamime) [inline src/components/switch/switch.stories.mdx:374]: This should be in separate `docs` commit.

File: src/components/table/table.component.js [hunk 5]
```diff
@@ -21,8 +21,8 @@ let deprecatedWarnTriggered = false;
 class Table extends React.Component {
   static contextType = LocaleContext;
 
-  constructor(props) {
-    super(props);
+  constructor(props, context) {
```
Comment (jamime) [inline src/components/table/table.component.js:24]: This should be in separate commit. Does this add any functionality or is it just code style?

**Review by jamime — CHANGES_REQUESTED** [review]

File: src/components/decimal/decimal.component.js (hunk 1 above)
Comment (mmaciorowski-sage) [reply src/components/decimal/decimal.component.js]: good point, changed, thanks

File: src/components/decimal/decimal.spec.js (hunk 3 above)
Comment (mmaciorowski-sage) [reply src/components/decimal/decimal.spec.js]: good point, changed, thanks

File: src/components/switch/switch.stories.mdx (hunk 4 above)
Comment (mmaciorowski-sage) [reply src/components/switch/switch.stories.mdx]: changed, thanks

File: src/components/configurable-items/configurable-items.component.js (hunk 2 above)
Comment (mmaciorowski-sage) [reply src/components/configurable-items/configurable-items.component.js]: just code style, we can ignore it for now

File: src/components/table/table.component.js (hunk 5 above)
Comment (mmaciorowski-sage) [reply src/components/table/table.component.js]: just code style, we can ignore it for now

File: src/components/decimal/decimal.spec.js [hunk 6]
```diff
@@ -1349,6 +1350,30 @@ describe("Decimal", () => {
...
+            <Decimal />
+          </I18nProvider>
+        );
+        expect(value()).toBe("0,00");
+        wrapper.setProps({
+          locale: {
+            locale: () => "en-GB",
+          },
+        });
+        wrapper.find('input[data-element="input"]').simulate("blur");
```
Comment (jamime) [inline src/components/decimal/decimal.spec.js:1372]: There is already a `blur()` method you can use.

**Review by jamime — CHANGES_REQUESTED** [review]

File: src/components/decimal/decimal.spec.js (hunk 6 above)
Comment (mmaciorowski-sage) [reply src/components/decimal/decimal.spec.js]: updated, thanks

**Review by jamime — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]
